vnext/Microsoft.ReactNative/Modules/AlertModule.cpp

```diff
@@ -98,6 +98,13 @@ void Alert::ProcessPendingAlertRequests() noexcept {
       });
 
       dialog.Closed([=](auto &&, auto &&) { xamlRoot.Changed(rootChangedToken); });
+    } else if (IsXamlIsland()) {
+      // We cannot show a ContentDialog in a XAML Island unless it is assigned a
+      // XamlRoot instance. In such cases, we just treat the alert as dismissed.
+      jsDispatcher.Post([result, this] { result(m_constants.dismissed, m_constants.buttonNeutral); });
+      pendingAlerts.pop();
+      ProcessPendingAlertRequests();
+      return;
```

Comment:
Where does the crash actually occur? This `return` prevents the code from getting further, but I don't see how this behavior change is any different for people who call this code early without having set xamlroot as @acoates-ms indicated.

Reply:
Crashes here: https://github.com/microsoft/react-native-windows/pull/11168/files#diff-3c81c385c361994bfc625387139150a42c69fc47f4b141648b55a327b8c54dacR128

Basically, you must set a XamlRoot before trying to show any alerts.

We could leave them queued indefinitely, but that seems like a race /  hang waiting to happen. E.g., imagine an app that creates a Promise from the `Alert.alert` call that gets resolved when a button is pressed or rejected when canceled or dismissed. Such apps would hang indefinitely.

Reply:
As an alternative, would it make sense to take the implementation I added in https://github.com/microsoft/react-native-windows/pull/11218 and fallback to that if there isn't a XamlRoot?

Reply:
@acoates-ms wouldn't that just run into the same issue when the top-level HWND is not set?